checks/evaluation/license.go

```diff
@@ -15,81 +15,91 @@
 package evaluation
 
 import (
+	"strings"
+
 	"github.com/ossf/scorecard/v4/checker"
 	sce "github.com/ossf/scorecard/v4/errors"
 	"github.com/ossf/scorecard/v4/finding"
+	"github.com/ossf/scorecard/v4/probes/hasApprovedLicense"
+	"github.com/ossf/scorecard/v4/probes/hasLicenseFile"
+	"github.com/ossf/scorecard/v4/probes/hasLicenseFileAtTopDir"
 )
 
-func scoreLicenseCriteria(f *checker.LicenseFile,
+// License applies the score policy for the Fuzzing check.
+func License(name string,
+	findings []finding.Finding,
 	dl checker.DetailLogger,
-) int {
-	var score int
-	msg := checker.LogMessage{
-		Path:   "",
-		Type:   finding.FileTypeNone,
-		Text:   "",
-		Offset: 1,
-	}
-	msg.Path = f.File.Path
-	msg.Type = finding.FileTypeSource
-	// #1 a license file was found.
-	score += 6
-
-	// #2 the licence was found at the top-level or LICENSE/ folder.
-	switch f.LicenseInformation.Attribution {
-	case checker.LicenseAttributionTypeAPI, checker.LicenseAttributionTypeHeuristics:
-		// both repoAPI and scorecard (not using the API) follow checks.md
-		// for a file to be found it must have been in the correct location
-		// award location points.
-		score += 3
-		msg.Text = "License file found in expected location"
-		dl.Info(&msg)
-		// for repo attribution prepare warning if not an recognized license"
-		msg.Text = "Any licence detected not an FSF or OSI recognized license"
-	case checker.LicenseAttributionTypeOther:
-		// TODO ascertain location found
-		score += 0
-		msg.Text = "License file found in unexpected location"
-		dl.Warn(&msg)
-		// for non repo attribution not the license detection is not supported
-		msg.Text = "Detecting license content not supported"
-	default:
-	}
-
-	// #3 is the license either an FSF or OSI recognized/approved license
-	if f.LicenseInformation.Approved {
-		score += 1
-		msg.Text = "FSF or OSI recognized license"
-		dl.Info(&msg)
-	} else {
-		// message text for this condition set above
-		dl.Warn(&msg)
+) checker.CheckResult {
+	// We have 3 unique probes, each should have a finding.
+	expectedProbes := []string{
+		hasLicenseFile.Probe,
+		hasApprovedLicense.Probe,
+		hasLicenseFileAtTopDir.Probe,
```

Comment:
do we really need a probe for this? Wouldn't file info be present in the `hasLicense` finding?

Reply:
for now, we'll leave as all 3. We can revisit path filtering later, may be applicable for binary artifact filtering etc. Will create issue

Reply:
Issue created: https://github.com/ossf/scorecard/issues/3590